### ISSUE TITLE
Can I use SAM3 to perform video tracking on tensor input?

### ISSUE THREAD

**Issue by lihaoyun6** [ultralytics/ultralytics#23744]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

I encountered an issue with `assert predictor.dataset.mode == "video"`  
I think this is because ultralytics currently only determines dataset mode based on the type of input, meaning that even if I input a video tensor, it will only treat it as a list of images.

### Use case

In my workflow, I have to encode videos into tensors to feed them into other models.
So if ultralytics could support video tensor inputs, it would avoid a lot of repetitive decoding time.

### Example
```python
from ultralytics.models.sam import SAM3VideoSemanticPredictor
overrides = dict(conf=0.45, task="segment", mode="predict",  model="./sam3.pt", half=True)
predictor = SAM3VideoSemanticPredictor(overrides=overrides)
results = predictor(source=video_tensor, text=["band"], stream=True)
for r in results:
    _ = r.masks
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lihaoyun6, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 Bug Report/behavior issue around `assert predictor.dataset.mode == "video"` when passing a tensor input, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that shows the tensor shape/type you’re passing, the exact call you’re making (SAM3 video tracking entrypoint), and the full traceback/logs. This will help us confirm whether this is expected behavior vs. a missing feature in input mode detection.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

This is an automated response 🤖—an Ultralytics engineer will also assist soon to dig into the video-mode assertion and discuss best next steps ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
What does video tensor contain? Is it a batch of tensors?

**Comment by lihaoyun6** [comment]
> What does video tensor contain? Is it a batch of tensors?

A 4D tensor of shape BHWC or BCHW, such as a tensor of [300, 720, 1280, 3] or [300, 3, 720, 1280], represent a 300-frame 720p video.

**Comment by glenn-jocher** [comment]
`SAM3VideoSemanticPredictor` currently expects `source` to resolve to a *video* dataloader (`predictor.dataset.mode == "video"` during `init_state()`, and per-frame indexing via `self.dataset.frame` in `inference()`), so a 4D BHWC/BCHW tensor passed via `source=` will be treated like an image batch and won’t run video tracking yet; details are in the [SAM `predict.py` reference](https://docs.ultralytics.com/reference/models/sam/predict/). Could you paste the full traceback and also run:

```bash
yolo checks
```

and confirm the tensor `dtype`, `device` (CPU/GPU), and whether it’s contiguous, so we can properly scope adding a tensor-backed video dataloader for this use case?
